command/pr_create.go

```diff
@@ -15,6 +15,39 @@ import (
 	"github.com/spf13/cobra"
 )
 
+type defaults struct {
+	Title string
+	Body  string
+}
+
+func computeDefaults(baseRef, headRef string) (defaults, error) {
+	commits, err := git.Commits(baseRef, headRef)
+	if err != nil {
+		return defaults{}, err
+	}
+
+	out := defaults{}
+
+	if len(commits) == 1 {
+		out.Title = commits[0].Title
+		body, err := git.CommitBody(commits[0].Sha)
+		if err != nil {
+			return defaults{}, err
+		}
+		out.Body = body
+	} else {
+		out.Title = headRef // TODO format or something?
```

Comment:
On dotcom we "humanize" the head ref by replacing underscores and dashes with spaces and transforming the whole string into sentence case

```
  #   "less_debris" => "Less debris"
  #   "rails-2.2.3" => "Rails 2.2.3"
  #   "various_fixes_and_stuff" => "Various fixes and stuff"
```

Reply:
i really dislike this feature and pointedly left it out D: i've found in practice that branch names are generally not suitable as humanized titles and trying to contort them into such is worse than just leaving them alone.

@ampinsk do you have an opinion? I can implement this if there is a concerted desire for it.

Reply:
How is it working right now? If it's already filling with the branch name, then I think my preference would be to "sentence-ize" it. After that I think my preference would be to use nothing

Reply:
@vilmibm I understood your `TODO` code comment as a note that you might be open to implementing parity with dotcom.

I don't feel so strongly about this, but I _would_ prefer eliminating dashes and underscores from the pre-filled PR title simply because I think they wouldn't look really great there if submitted as-is. Also, that brings us consistency with the web flow behavior.